crypto/txscript/src/opcodes/mod.rs

```diff
@@ -531,10 +552,57 @@ opcode_list! {
     }
 
     // Bitwise logic opcodes.
-    opcode OpInvert<0x83, 1>(self, vm) Err(TxScriptError::OpcodeDisabled(format!("{self:?}")))
-    opcode OpAnd<0x84, 1>(self, vm) Err(TxScriptError::OpcodeDisabled(format!("{self:?}")))
-    opcode OpOr<0x85, 1>(self, vm) Err(TxScriptError::OpcodeDisabled(format!("{self:?}")))
-    opcode OpXor<0x86, 1>(self, vm) Err(TxScriptError::OpcodeDisabled(format!("{self:?}")))
+    opcode OpInvert<0x83, 1>(self, vm){
+        if vm.flags.covenants_enabled{
+            let data = vm.dstack.pop()?;
+            let r: Vec<u8> = data.iter().map(|b| !b).collect();
```

Comment:
It's possible to update dats in place `data.iter_mut().for_each(|b| *b = !*b);`
And push it in stack